lib/rules/property-sort-order.js

```diff
@@ -110,35 +95,51 @@ module.exports = {
           if (name) {
             if (parser.options['ignore-custom-properties']) {
               if (propertyCheckList.indexOf(name.content) !== -1) {
-                properties[name.content] = prop;
+                properties.push({
+                  name: name.content,
+                  line: prop.start.line,
+                  column: prop.start.column
+                });
               }
             }
             else {
-              properties[name.content] = prop;
+              properties.push({
+                name: name.content,
+                line: prop.start.line,
+                column: prop.start.column
+              });
             }
           }
+
+
         });
 
+        // console.log('properties', properties);
+
+        // console.log('order', order);
+
         sorted = sortProperties(properties, order);
 
-        pKeys = Object.keys(properties);
-        sKeys = Object.keys(sorted);
-
-        sKeys.every(function (e, i) {
-          var pKey = pKeys[i],
-              prop = properties[pKey];
-
-          if (e !== pKey) {
-            result = helpers.addUnique(result, {
-              'ruleId': parser.rule.name,
-              'line': prop.start.line,
-              'column': prop.start.column,
-              'message': 'Expected `' + e + '`, found `' + pKey + '`',
-              'severity': parser.severity
-            });
-          }
-          return true;
-        });
+        expectedOrder = sorted.map(function (prop) {
+          return prop.name;
+        }).join(', ');
+
+        currentOrder = properties.map(function (prop) {
+          return prop.name;
+        }).join(', ');
+
+        // console.log('expectedOrder', expectedOrder);
+        // console.log('currentOrder', currentOrder);
```

Comment:
@benthemonkey It passes ESLint and there is no mention of this requirement in https://github.com/sasstools/sass-lint/blob/develop/CONTRIBUTING.md. It is left there for the convenience of the potential contributors.